packages/react-reconciler/src/__tests__/ReactUpdaters-test.internal.js

```diff
@@ -0,0 +1,603 @@
+      expect(allSchedulerTypes[1]).toContain(HighPriorityUpdater);
+
+      // Finish the initial partial update
+      triggerLowPriorityUpdate();
+      expect(Scheduler).toFlushAndYieldThrough([
+        'LowPriorityUpdater 2',
+        'Yield LowPriority 2',
+        'onCommitRoot',
+      ]);
+    });
+    expect(allSchedulerTypes).toHaveLength(3);
+    expect(allSchedulerTypes[2]).toHaveLength(1);
+    expect(allSchedulerTypes[2]).toContain(LowPriorityUpdater);
+
+    // Verify no outstanding flushes
+    Scheduler.unstable_flushAll();
+  });
+
+  // @gate experimental
+  it('should not lose track of updaters if work yields before finishing', async () => {
+    const {HostRoot} = require('react-reconciler/src/ReactWorkTags');
+
+    const Yield = ({renderTime}) => {
+      Scheduler.unstable_advanceTime(renderTime);
+      Scheduler.unstable_yieldValue('Yield:' + renderTime);
+      return null;
+    };
+
+    let first;
+    class FirstComponent extends React.Component {
+      state = {renderTime: 1};
+      render() {
+        first = this;
+        Scheduler.unstable_advanceTime(this.state.renderTime);
+        Scheduler.unstable_yieldValue(
+          'FirstComponent:' + this.state.renderTime,
+        );
+        return <Yield renderTime={4} />;
+      }
+    }
+    let second;
+    class SecondComponent extends React.Component {
+      state = {renderTime: 2};
+      render() {
+        second = this;
+        Scheduler.unstable_advanceTime(this.state.renderTime);
+        Scheduler.unstable_yieldValue(
+          'SecondComponent:' + this.state.renderTime,
+        );
+        return <Yield renderTime={7} />;
+      }
+    }
+
+    Scheduler.unstable_advanceTime(5); // 0 -> 5
+
+    const root = ReactDOM.unstable_createRoot(document.createElement('div'));
+    root.render(
+      <React.Fragment>
+        <FirstComponent />
+        <SecondComponent />
+      </React.Fragment>,
+    );
+
+    // Render everything initially.
+    expect(Scheduler).toFlushAndYield([
+      'FirstComponent:1',
+      'Yield:4',
+      'SecondComponent:2',
+      'Yield:7',
+      'onCommitRoot',
+    ]);
+    expect(allSchedulerTags).toHaveLength(1);
+    expect(allSchedulerTags[0]).toHaveLength(1);
+    expect(allSchedulerTags[0]).toContain(HostRoot);
+
+    // Render a partial update, but don't finish.
+    first.setState({renderTime: 10});
+    expect(Scheduler).toFlushAndYieldThrough(['FirstComponent:10']);
+    expect(allSchedulerTypes).toHaveLength(1);
+
+    // Interrupt with higher priority work.
+    ReactDOM.flushSync(() => second.setState({renderTime: 30}));
+    expect(Scheduler).toHaveYielded([
+      'SecondComponent:30',
+      'Yield:7',
+      'onCommitRoot',
+    ]);
+    expect(allSchedulerTypes).toHaveLength(2);
+    expect(allSchedulerTypes[1]).toHaveLength(1);
+    expect(allSchedulerTypes[1]).toContain(SecondComponent);
+
+    // Resume the original low priority update.
+    expect(Scheduler).toFlushAndYield([
```

Comment:
Seems to me that this is the only difference to the [prior test ("should distinguish between updaters in the case of interleaved work")](https://github.com/facebook/react/pull/21171/files#diff-4962849261d085a5fbf4389513ab1a3a9b442cc8e1c4629f130ba2e03ef2b6bfR446):
```diff
-     // Resume the original low priority update.
-     expect(Scheduler).toFlushAndYield([
+     // Finish the initial partial update
+     triggerLowPriorityUpdate();
+     expect(Scheduler).toFlushAndYieldThrough([
        'FirstComponent:10',
        'Yield:4',
        'onCommitRoot',
      ]);
```

Would it make sense to normalize these two tests (function vs class components, names etc) so that this difference is more obvious? It's not clear why the update is considered "low priority" in the prior test but "partial" in the current test. Aren't both partial and low priority?

Reply:
Yeah, you're right. These overlap a ton. I should just remove one of them.

`toFlushAndYieldThrough` could be `toFlushAndYield` here. That's the full set of remaining pending stuff.